packages/sdk/src/adapters/multipeer/session.ts

```diff
@@ -166,10 +166,27 @@ export class Session extends EventEmitter {
 			// tslint:disable-next-line:no-console
 			log.error('network', `[ERROR] setAuthoritativeClient: client ${clientId} does not exist.`);
 		}
-		this._clientSet[clientId].setAuthoritative(true);
-		this.clients
-			.filter(client => client.id !== clientId && client.isJoined())
-			.forEach(client => client.setAuthoritative(false));
+		const oldAuthority = this.authoritativeClient;
+		const newAuthority = this._clientSet[clientId];

```

Comment:
I was hoping this could be handled on pipe.remote. Getting the stats from the authoritative peer is a lot of new code.

Reply:
Nothing gets serialized or deserialized anywhere but in a WebSocket connection. If we did it in the pipe it would add overhead.

---
In reply to: [325909690](https://github.com/microsoft/mixed-reality-extension-sdk/pull/445#discussion_r325909690) [](ancestors = 325909690)

Reply:
Agreed this is better approach than adding overhead in in pipe.remote.

---
In reply to: [325918122](https://github.com/microsoft/mixed-reality-extension-sdk/pull/445#discussion_r325918122) [](ancestors = 325918122,325909690)